hathor/utils/api.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+from typing import List, Optional, TypeVar
+
+from pydantic import Field, ValidationError, validator
+from twisted.web.http import Request
+
+from hathor.api_util import get_args
+from hathor.utils.pydantic import BaseModel
+
+T = TypeVar('T')
+
+
+def _list_to_single_item(v: List[T]) -> Optional[T]:
```

Comment:
Missing docstring.

Reply:
Done! Also, I've moved it to a `List` utils and added tests for it.

By the way, I'm not sure if you noticed but I've been populating a new `hathor/utils` package with separate files, to better namespace util functions and keep thinks more organized and modular. I think in the future we could slowly migrate functions from `hathor/util.py` to more specific utils files in this `utils` package.

Reply:
Thanks!

I noticed your move. That's a good one. We can refactor `hathor/util.py` on-demand.